frontend/src/components/SamplesList.tsx

```diff
@@ -262,10 +265,25 @@ export default function SamplesList({
       {showDownloadModal && (
         <DownloadModal
           loader={() => {
-            return Promise.resolve(buildTsvString(samples!, columnDefs));
+            return sampleCount <= MAX_ROWS_TABLE
+              ? Promise.resolve(buildTsvString(samples!, columnDefs))
+              : refetch({
+                  options: {
+                    limit: MAX_ROWS_EXPORT,
+                  },
+                }).then((result) =>
+                  buildTsvString(result.data.samples, columnDefs)
+                );
```

Comment:
If the already-downloaded rows are <= 500 rows, we can just download them directly without refetching.